### DISCUSSION TITLE
rain values in forecast data

### DISCUSSION THREAD

**Discussion by ttuning** [weather-gov/api#600] [General]
I need the forecasted rain amount in the forecast data.   Is there a way to retrieve that?   Can't find it in the doc.   Maybe a Feature-Flag?

**Comment by jmb1304** [comment]
This is something that I am trying to figure out as well. Unfortunately, it seems as though the feature has been implemented but not released yet [https://github.com/weather-gov/api/discussions/368](https://github.com/weather-gov/api/discussions/368)

**Comment by JJKraw** [comment]
AFAIK the only place rainfall _amount_ is mentioned is in the detailedForecast fields of the point forecast. Text along the lines of "New rainfall amounts between a tenth and quarter of an inch possible." Note that expected snowfall _is_ available in the gridpoints/x/y/ endpoint.

**Comment by netbymatt** [comment]
@JJKraw That is a very interesting distinction that I'm just realizing now. I've been using probability of precipitation for the longest time and just mentally correlating that to heavy rain when it reaches >90%. But yes, snow and ice both show actual amounts in inches/mm and rain does not.

From experience I have learned from years of reading this data and correlating it with text forecast products that there's a trick to reading those values. For snow you might see a timestamp with a 6 hour duration. You should read this (and present it to the end user) as "Over the 6 hours starting at 2pm 4.5 inches of snow will fall." It does not mean 4.5 inches of snow per hour.

**Comment by sullynole** [comment]
Use the points endpoint to find the url for the forecastGridData where you will find the quantitativePrecipitation field.

**Reply by netbymatt** [reply]
I completely agree this data is there. But I feel until just recently this field has been present but always null.

I checked several locations where rain is forecasted for the next week and it's there for all of them. For everyone's benefit here's an excerpt from https://api.weather.gov/gridpoints/HUN/48,44 at the current date/time.
```
quantitativePrecipitation : 
	uom : "wmoUnit:mm",
	values : [
		0 : {validTime: "2022-12-02T09:00:00+00:00/PT3H", value: 0},
		1 : {validTime: "2022-12-02T12:00:00+00:00/PT6H", value: 0},
		2 : {validTime: "2022-12-02T18:00:00+00:00/PT6H", value: 0},
		3 : {validTime: "2022-12-03T00:00:00+00:00/PT6H", value: 1.016},
		4 : {validTime: "2022-12-03T06:00:00+00:00/PT6H", value: 5.334},
		5 : {validTime: "2022-12-03T12:00:00+00:00/PT6H", value: 2.286},
```

Looks like I have a weekend project updating some code in my app!

**Reply by ttuning** [reply]
At the time of this writing 2022-12-03 is in the future.  So I am trying to figure out what line 3 is telling me above.  Does PT6H mean that there will be 1.016  mm? inches? of rain between 6pm and midnight?

**Reply by JJKraw** [reply]
The uom field indicates mm and that makes way more sense than inches. Yes on the latter question: amount of precip during the range given.

Also some coding in store for me :).

**Reply by sullynole** [reply]
> 3 : {validTime: "2022-12-03T00:00:00+00:00/PT6H", value: 1.016}

for the 6 hour time period (/PT6H) starting at 2022-12-03 00:00:00 UTC (2022-12-03T00:00:00+00:00) there will be 1.016 mm of precipitation.  The unit of measurement for all values is defined above the values array.

**Comment by ttuning** [comment]
Thanks for helping me understanding the values.  However, I am trying to write algorithms that predict flooding based on forecasted rain amounts as well as water level sensors placed in streams.   3 hour and 6 hour forecast windows are pretty sparce to make accurate predictions.  So my question is, Are the forecasts at the hourly level not available at all or just not surfaced in the API?   When I call the MicroSoft maps API it gives me this data back in an hourly format for the next 12 hours.   However, I am not sure I trust these values because, how does MS make these forecasts?   They don't have weather stations that I imagine.   I am assuming they are just calling some government weather data and making calculations based on the results.  So that I what I am trying to do as well.  Skipping the middle man.   Any help is greatly appreciated.

**Reply by sullynole** [reply]
> Thanks for helping me understanding the values. However, I am trying to write algorithms that predict flooding based on forecasted rain amounts as well as water level sensors placed in streams. 3 hour and 6 hour forecast windows are pretty sparce to make accurate predictions. So my question is, Are the forecasts at the hourly level not available at all or just not surfaced in the API? When I call the MicroSoft maps API it gives me this data back in an hourly format for the next 12 hours. However, I am not sure I trust these values because, how does MS make these forecasts? They don't have weather stations that I imagine. I am assuming they are just calling some government weather data and making calculations based on the results. So that I what I am trying to do as well. Skipping the middle man. Any help is greatly appreciated.

API gridpoints endpoints are exposing the forecast grids created at local NWS weather forecast offices curated by local forecasters.  While forecasters typically utilize models to aid in creating their forecasts, the API does not contain pure model data and local/regional policies guide the elements, intervals, and update frequency of forecast grids.  

It's possible other applications are using models, either private or public. Raw model output from the NWS can be found here: https://nomads.ncep.noaa.gov
